config_registry.go

```diff
@@ -0,0 +1,82 @@
+package evergreen
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/mongodb/grip"
+)
+
+var registry *configSectionRegistry
+
+func init() {
+
+	if err := resetRegistry(); err != nil {
+		panic(fmt.Sprintf("error registering config sections: %s", err.Error()))
+	}
+}
+
+type configSectionRegistry struct {
+	mu       *sync.RWMutex
```

Comment:
I would not make this a pointer, so you can avoid needing to explicitly construct this value, because the registry.